src/jaxsim/parsers/kinematic_graph.py

```diff
@@ -971,13 +971,10 @@ def find_parent_link_of_frame(self, name: str) -> str:
         except KeyError as e:
             raise ValueError(f"Frame '{name}' not found in the kinematic graph") from e
 
-        match frame.parent.name:
-            case parent_name if parent_name in self.graph.links_dict:
-                return parent_name
-
-            case parent_name if parent_name in self.graph.frames_dict:
-                return self.find_parent_link_of_frame(name=parent_name)
-
-            case _:
-                msg = f"Failed to find parent element of frame '{name}' with name '{frame.parent.name}'"
-                raise RuntimeError(msg)
+        if frame.parent_name in self.graph.links_dict:
+            return frame.parent_name
+        elif frame.parent_name in self.graph.frames_dict:
+            return self.find_parent_link_of_frame(name=frame.parent_name)
+        else:
+            msg = f"Failed to find parent element of frame '{name}' with name '{frame.parent_name}'"
+            raise RuntimeError(msg)
```

Comment:
The `else` can be removed
```suggestion
        if frame.parent_name in self.graph.links_dict:
            return frame.parent_name
        elif frame.parent_name in self.graph.frames_dict:
            return self.find_parent_link_of_frame(name=frame.parent_name)

        msg = f"Failed to find parent element of frame '{name}' with name '{frame.parent_name}'"
        raise RuntimeError(msg)
```